extensions/iceberg/src/main/java/io/deephaven/iceberg/location/IcebergTableLocationKey.java

```diff
@@ -4,67 +4,22 @@
 package io.deephaven.iceberg.location;
 
 import io.deephaven.engine.table.impl.locations.TableLocationKey;
-import io.deephaven.engine.table.impl.locations.local.URITableLocationKey;
-import io.deephaven.parquet.table.ParquetInstructions;
 import io.deephaven.parquet.table.location.ParquetTableLocationKey;
-import org.apache.iceberg.FileFormat;
-import org.jetbrains.annotations.NotNull;
-import org.jetbrains.annotations.Nullable;
-
-import java.net.URI;
-import java.util.Map;
 
 /**
- * {@link TableLocationKey} implementation for use with data stored in the parquet format.
+ * {@link TableLocationKey} implementation for use with data stored in Iceberg tables.
  */
-public class IcebergTableLocationKey extends URITableLocationKey {
-
-    private static final String IMPLEMENTATION_NAME = IcebergTableLocationKey.class.getSimpleName();
-
-    final FileFormat format;
-    final URITableLocationKey internalTableLocationKey;
-
+public interface IcebergTableLocationKey extends TableLocationKey {
     /**
-     * Construct a new IcebergTableLocationKey for the supplied {@code fileUri} and {@code partitions}.
+     * Get the read instructions for the location.
      *
-     * @param fileUri The file that backs the keyed location.
-     * @param order Explicit ordering index, taking precedence over other fields
-     * @param partitions The table partitions enclosing the table location keyed by {@code this}. Note that if this
-     *        parameter is {@code null}, the location will be a member of no partitions. An ordered copy of the map will
-     *        be made, so the calling code is free to mutate the map after this call
-     * @param readInstructions the instructions for customizations while reading
+     * @return the read instructions
      */
-    public IcebergTableLocationKey(
-            final FileFormat format,
-            @NotNull final URI fileUri,
-            final int order,
-            @Nullable final Map<String, Comparable<?>> partitions,
-            @NotNull final Object readInstructions) {
-        super(fileUri, order, partitions);
-        this.format = format;
-        if (format == FileFormat.PARQUET) {
-            // This constructor will perform validation of the parquet file
-            final ParquetInstructions parquetInstructions = (ParquetInstructions) readInstructions;
-            this.internalTableLocationKey =
-                    new ParquetTableLocationKey(fileUri, order, partitions, parquetInstructions);
-        } else {
-            throw new IllegalArgumentException("Unsupported file format: " + format);
-        }
-    }
-
-    @Override
-    public String getImplementationName() {
-        return IMPLEMENTATION_NAME;
-    }
+    Object readInstructions();
 
     /**
-     * See {@link ParquetTableLocationKey#verifyFileReader()}.
+     * Verify that a reader for the file can be created successfully. Delegates to
+     * {@link ParquetTableLocationKey#verifyFileReader()} for parquet files.
```

Comment:
I'm not sure the implementation detail belongs in the interface JavaDoc.
Also, should we have a more generic name, like "verifyExistence()" or "validate()" or something?

Reply:
Currently this matches the underlying `ParquetTableLocationKey` method